pkg/gadgets/internal/networktracer/tracer.go

```diff
@@ -116,20 +136,30 @@ func (t *Tracer[Event]) newAttachment(
 }
 
 func NewTracer[Event any](
-	spec *ebpf.CollectionSpec,
-	bpfProgName string,
-	bpfPerfMapName string,
 	baseEvent func(ev types.Event) *Event,
 	processEvent func(rawSample []byte, netns uint64) (*Event, error),
 ) (_ *Tracer[Event], err error) {
-	gadgets.FixBpfKtimeGetBootNs(spec.Programs)
-
 	t := &Tracer[Event]{
 		attachments:  make(map[uint64]*attachment),
 		baseEvent:    baseEvent,
 		processEvent: processEvent,
 	}
 
+	dispatcherTailCallMapSpec, err := loadDispatcherTailCallMap()
+	if err != nil {
+		return nil, err
+	}
+	var opts ebpf.CollectionOptions
+	if err = dispatcherTailCallMapSpec.LoadAndAssign(&t.dispatcherTailCallMapObjs, &opts); err != nil {
+		return nil, fmt.Errorf("loading ebpf program: %w", err)
+	}
+
+	return t, nil
+}
```

Comment:
It can work but it means the map parameter (`max_entries`) are defined at 2 different places and need to be kept in sync. However, that's just one parameter, and your solution reduces the amount of files. I'm ok either way. What do you prefer?

Reply:
I think it's better to reduce the amount of files. We can add a comment in both places to keep the parameter in sync.

Reply:
Ok, I am removing the file and I added the 2 comments